Feat: Scroll To Top Button

ParthJohri opened the pull request:
# Scroll To Top Button [Issue No: 1](https://github.com/monacodelisa/LGBTQ-definitions/issues/1)

## Changes

1. Added Scroll to button HTML file which is making use of the font awesome toolkit.
2. Imported font awesome in the `index.html` file.
3. Added the Scroll to button component, which considered the page length while smooth scrolling to the top.
4. Added Scroll to button scss file for the desktop and mobile CSS file of the scroll to top button.
5. Updated the app component, to import the scroll to the top button.

## Description

## What type of PR is this? (check all applicable)

- [x] ✨ Feature

## Mobile & Desktop Screenshots/Recordings
![screenshot-scroll-1](https://github.com/monacodelisa/LGBTQ-definitions/assets/76129377/dc73078b-5621-44c4-bb1b-db0cc6011298)
![screenshot-scroll-2](https://github.com/monacodelisa/LGBTQ-definitions/assets/76129377/f9195929-ad5a-407c-bed1-e69e20df1b3e)
![screenshot-scroll-3](https://github.com/monacodelisa/LGBTQ-definitions/assets/76129377/bf4741f0-1942-4233-8df4-2747dc2c33d0)
![screenshot-scroll-4](https://github.com/monacodelisa/LGBTQ-definitions/assets/76129377/f580df74-3fc1-464d-8011-0b942481e8aa)

monacodelisa commented on src/app/app.component.html:
```diff
@@ -1,3 +1,4 @@
-<app-footer/>
+<app-header />
+<router-outlet />
+<app-scroll-to-top />
```
there is no need for a separate component

monacodelisa commented on src/app/components/scroll-to-top/scroll-to-top.component.html:
```diff
@@ -0,0 +1,5 @@
+<div class="scroll-to-top">
+  <button type="button" class="btn active" (click)="scrollToTop()">
+    <i class="fas fa-chevron-up"></i>
+  </button>
```
since the project already uses `PrimeNG` including `primeicons` we do not need to deal with `font-awesome` at all

monacodelisa commented on src/app/components/scroll-to-top/scroll-to-top.component.ts:
```diff
@@ -0,0 +1,26 @@
+  isVisible = false;
+
+  @HostListener('window:scroll', [])
+  onWindowScroll() {
```
this part is good, this can be used with some minor adjustments

monacodelisa commented on src/index.html:
```diff
@@ -1,23 +1,27 @@
+    <link
+      rel="stylesheet"
+      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css"
+    />
```
no need for this

monacodelisa commented on src/index.html:
```diff
@@ -1,23 +1,27 @@
+    <meta
+      property="og:description"
+      content="LGBTQ+ Terms & Definitions is an open source project created to provide clear and accurate definitions of LGBTQ+ terms. This educational resource aims to foster understanding and inclusion."
+    />
```
please try to exclude unnecessary indentation changes, if you are using vs code you can select specific sections and `revert  selected range` so it is not part of the commit

monacodelisa reviewed (changes requested):
@ParthJohri the visual result is very accurate, just the implementation can be done in a better way, two key points: 

- no need for separate component, logic can be applied directly in the `app.component.ts` I believe the description mentioned this 
- no need for font-awesome, they are a nice chouse unless there is a better one :) and for this case `pribmeicons` are a better choice because they are already available and have exactly the icon needed 

here is a link to reference an existing implementation of the exact same functionality 

[app.component.ts](https://github.com/AnguHashBlog/angular-primeng/blob/a0fb57991e35377e4b3b320d004e7ecc2c637ea0/src/app/app.component.ts#L58)

[app.component.html](https://github.com/AnguHashBlog/angular-primeng/blob/a0fb57991e35377e4b3b320d004e7ecc2c637ea0/src/app/app.component.html#L3)

[app.component.scss](https://github.com/AnguHashBlog/angular-primeng/blob/a0fb57991e35377e4b3b320d004e7ecc2c637ea0/src/app/app.component.scss#L1)

after reviewing the files you can replicate this implementation, let me know if you have further question 

good job all and all, it can only get better :)

ParthJohri commented:
@monacodelisa I have resolved all the changes.

## Changes 

1. Removed scroll-to-top component folder.
2. Updated the app.component to imitate the scroll to top feature.
3. Removed the indentations and import font awesome toolkit instead used primeng p-button.
4. Used window's scroll behaviour for smooth scrolling.

However over here I implemented the scroll option by using window's scroll method instead of viewportscroller since the later was not having any option of smooth behaviour.

## Via `Window Scroll` Method

https://github.com/monacodelisa/LGBTQ-definitions/assets/76129377/36a38d2e-6c12-415b-b2e0-e8cff5390f5c

## Via `ViewScroller` Method

https://github.com/monacodelisa/LGBTQ-definitions/assets/76129377/d3be9267-fc3a-41d4-9a0b-72e289e81da8

Due to the abrupt behaviour, I implemented the above `window scroll` method.

Please let me know if this works, and if there any other changes to be made.

monacodelisa reviewed (approved):
@ParthJohri cool :) that looks great thank you for the contribution
